pkg/server/endpoints/node/handler.go

```diff
@@ -53,17 +55,31 @@ type Handler struct {
 	c       HandlerConfig
 	limiter Limiter
 
-	dsCache *datastoreCache
+	dsCache                          *datastoreCache
+	fetchRegistrationEntriesCache    *regentryutil.FetchRegistrationEntriesCache
+	useFetchRegistrationEntriesCache bool
 }
 
 func NewHandler(config HandlerConfig) *Handler {
 	if config.Clock == nil {
 		config.Clock = clock.New()
 	}
+	fetchX509SVIDCache, err := regentryutil.NewFetchX509SVIDCache(fetchSVIDCacheSize)
+	if err != nil {
+		config.Log.Errorf("could not create lru cache. Not using cache for FetchX509SVID", err)
+		return &Handler{
```

Comment:
Consider simplifying the logic to only return a `Handler` at the end of the method and having the supporting implementation conditionally initialize variables that are set to the field values in the built `Handler`.